src/tokensource.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Token Source implementation for [Azure Managed Identities](https://docs.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview).
+ */
+export interface ManagedIdentityTokenSource {
+    /** @hidden */
+    kind: "AzureManagedIdentity";
+
+    /**
+     * The Azure Active Directory resource identifier of the web API being invoked.
+     * For example, `https://management.core.windows.net/` or `https://graph.microsoft.com/`.
+     */
+    resource: string;
+}
+
+// Over time we will likely add more implementations
+export type TokenSource = ManagedIdentityTokenSource;
+
+/**
+ * Returns a `ManagedIdentityTokenSource` object.
+ * @param resource The Azure Active Directory resource identifier of the web API being invoked.
+ * @example Gets a `ManagedIdentityTokenSource` object.
+ * ```javascript
+ * const df = require("durable-functions");
+ *
+ * module.exports = df.orchestrator(function*(context) {
+ *     return yield context.df.callHttp(
+ *         "GET",
+ *         "https://management.azure.com/subscriptions?api-version=2019-06-01",
+ *         null,
```

Comment:
I believe this sample needs another `null` value for headers.

Also, as a NIT, I have heard [some recommendations](https://basarat.gitbooks.io/typescript/docs/javascript/null-undefined.html) to use `undefined` over `null`. Apparently the TypeScript team doesn't use `null` at all. However, `null` is all over Node, so this is likely fine.

Reply:
Good catch on the missing parameter.

I'll use `undefined` for both parameters instead of `null`. We seem to use `undefined` more frequently in this codebase than we do `null`, though it's not totally consistent.